src/main/scopes/npm/get-directory-prefix.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright IBM Corp. 2023, 2023
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import { type Logger } from '../../core/log/logger.js'
+import { runCommand } from '../../core/run-command.js'
+
+const cache = new Map<string, string>()
+
+/**
+ * Given a directory path, get the directory of the package it belongs to.
+ *
+ * @param dirPath - A directory to find containing package for. It may or may not include a
+ * package.json file directly in it. This is an absolute path.
+ * @param logger - Logger instance.
+ * @returns A string indicating the closest (ascendent) package directory.
+ */
+export async function getDirectoryPrefix(dirPath: string, logger: Logger): Promise<string> {
```

Comment:
General question: Is there ever a situation where this doesn't return a result for a given dir path? Or even if there's no node modules or package.json files or anything does it still return a directory prefix?

Reply:
looks like it returns the directory itself if there isn't one up the hierarchy 
<img width="437" alt="image" src="https://github.com/ibm-telemetry/telemetry-js/assets/40550942/ba497e97-6bd2-4b02-a7f3-4768807cff40">